util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java

```diff
@@ -131,12 +114,22 @@ private HealthStatus getHealth(final String componentName) {
 
     @Override
     public void onFailure() {
-      onComponentFailure(componentName);
+      actor.run(
+          () -> {
+            log.error("{} failed, marking it as unhealthy", componentName);
+            componentHealth.computeIfPresent(componentName, (k, v) -> HealthStatus.UNHEALTHY);
+            calculateHealth();
+          });
     }
```

Comment:
Nit: could we extract the closures to methods? e.g.

```java
@Override
public void onFailure() {
  actor.run(this::onComponentFailure);
}

private void onComponentFailure() {
  log.error("{} failed, marking it as unhealthy", componentName);
  componentHealth.computeIfPresent(componentName, (k, v) -> HealthStatus.UNHEALTHY);
  calculateHealth();
}
```

I have no strong arguments other than I find it more readable, but could just be me :sweat_smile:

Reply:
It was like this before. I don't know why I changed :grinning: